src/libsyntax/ext/expand.rs

```diff
@@ -454,7 +454,7 @@ impl<'a, 'b> MacroExpander<'a, 'b> {
         };
         let path = &mac.node.path;
 
-        let ident = ident.unwrap_or(keywords::Invalid.ident());
+        let ident = ident.unwrap_or_else(|| keywords::Invalid.ident());
```

Comment:
@petrochenkov I thought so, too, until I benchmarked a very simple expression (`_.unwrap_or..(vec![])`) with / without the `else`. Interestingly, though there's no allocation, and the result should be basically static, the closure version was a wee bit faster. Not that it matters in this place, so if you like, I can roll it back.